src/modules/simulator/simulator_mavlink.cpp

```diff
@@ -1124,15 +1124,33 @@ int Simulator::publish_distance_topic(const mavlink_distance_sensor_t *dist_mavl
 	dist.orientation = dist_mavlink->orientation;
 	dist.variance = dist_mavlink->covariance * 1e-4f; // cm^2 to m^2
 	dist.signal_quality = -1;
-
 	dist.h_fov = dist_mavlink->horizontal_fov;
 	dist.v_fov = dist_mavlink->vertical_fov;
 	dist.q[0] = dist_mavlink->quaternion[0];
 	dist.q[1] = dist_mavlink->quaternion[1];
 	dist.q[2] = dist_mavlink->quaternion[2];
 	dist.q[3] = dist_mavlink->quaternion[3];
 
-	_dist_pub.publish(dist);
+	// Create 1st publisher instance if it doesn't exist
+	if (_dist_pubs[0] == nullptr) {
+		_dist_pubs[0] = new uORB::PublicationMulti<distance_sensor_s> {ORB_ID(distance_sensor)};
+		_dist_sensor_ids[0] = dist.id;
+	}
+
+	// New publishers will be created based on the sensor ID's being different or not
+	for (size_t i = 0; i < sizeof(_dist_sensor_ids) / sizeof(_dist_sensor_ids[0]); i++) {
+		if (_dist_sensor_ids[i] == dist.id) {
```

Comment:
You need to add a pointer check, otherwise you risk dereferencing nulllpointers, in case `dist.id == 0`:
```suggestion
		if (_dist_pubs[i] && _dist_sensor_ids[i] == dist.id) {
```